src/temp_sensor.rs

```diff
@@ -136,9 +136,9 @@ pub struct TemperatureSensorDriver {
 }
 
 impl TemperatureSensorDriver {
-    pub fn new(config: TemperatureSensorConfig) -> Result<Self, EspError> {
+    pub fn new(config: &TemperatureSensorConfig) -> Result<Self, EspError> {
         let mut sensor = core::ptr::null_mut();
-        esp!(unsafe { temperature_sensor_install(&config.into(), &mut sensor) })?;
+        esp!(unsafe { temperature_sensor_install(&config.clone().into(), &mut sensor) })?;
```

Comment:
@ivmarkov is this what you had in mind with

> Take the config by & as we do everywhere else.

?

Reply:
Yes. To avoid the `clone()` though you need to change `From<TemperatureSensorConfig>` to `From<&TemperatureSensorConfig>`.

Also, I'm no longer sure who is driving it - is that you, or @lu-zero ? Ultimately, we of course want one single PR that we can merge, so the two of you should probably sync with each other etc.. :)

Reply:
> Yes. To avoid the clone() though you need to change From<TemperatureSensorConfig> to From<&TemperatureSensorConfig>.

Alright, will do that (later).

> Also, I'm no longer sure who is driving it - is that you, or @lu-zero ? 

Both of us I think.

> Ultimately, we of course want one single PR that we can merge, so the two of you should probably sync with each other etc.. :)

As far as I understood https://github.com/esp-rs/esp-idf-hal/pull/337#issuecomment-2336828281, @lu-zero is fine  with continuing here. I've reset my branch to his and have also given him write access to my repo so he can add to this PR.

(@lu-zero Let me know if I misunderstood something)

Reply:
> Yes. To avoid the clone() though you need to change From<TemperatureSensorConfig> to From<&TemperatureSensorConfig>.

I've encountered a problem with `value.clk_src` in the From trait impl. As far as I understood it, `into()` consumes the value instead of taking a reference. The solution I ended up with (https://github.com/esp-rs/esp-idf-hal/pull/337/commits/c061d00195bdb167e796cae140d763192774c84d) is to change the trait for `TempSensorClockSource` too. I'm relatively new to Rust, so let me know if there is another/better way.

Reply:
> > Yes. To avoid the clone() though you need to change From to From<&TemperatureSensorConfig>.
> 
> I've encountered a problem with `value.clk_src` in the From trait impl. As far as I understood it, `into()` consumes the value instead of taking a reference. The solution I ended up with ([c061d00](https://github.com/esp-rs/esp-idf-hal/commit/c061d00195bdb167e796cae140d763192774c84d)) is to change the trait for `TempSensorClockSource` too. I'm relatively new to Rust, so let me know if there is another/better way.

Yes - just mark `TempSensorClockSource` as `Copy`. It is a simple "C style" enum so marking these with `Copy` is always a good idea.